### PR TITLE
ci: build and deploy with GitHub Actions

### PR DESCRIPTION
**Pull request by lwille**
related to INFRA-1258

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        restore-keys: |
+          ${{ runner.os }}-maven-
+    - if: github.event.release
+      name: Update version in pom.xml (Release only)
+      run: mvn -B versions:set -DnewVersion=${{ github.event.release.tag_name }} -DgenerateBackupPoms=false
+    - name: Run Maven (${{ steps.release-detection.outputs.type }})
+      run: mvn -B clean install
+    - name: Publish package (${{ steps.release-detection.outputs.type }})
+      run: |-
+        echo -e "${MVN_CENTRAL_GPG_KEY_SEC}" gpg -q --allow-secret-key-import --import
```
Comment (maxdanilov) [inline .github/workflows/deploy.yml:73]: not sure I understand, where does this GPG thing come from?

Comment (lwille) [reply .github/workflows/deploy.yml]: uh, there was actually a pipe missing :see_no_evil:

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        asset_content_type: application/zip
+    - if: github.event.release
+      name: Checkout ${{ github.event.repository.default_branch }} and commit new development version (Release only)
+      run: |-
+        git config --global user.name "Release Bot"
+        git config --global user.email actions@github.com
+        git fetch
+        git checkout -b ${{ github.event.repository.default_branch }} --track origin/${{ github.event.repository.default_branch }}
+        mvn -B org.apache.maven.plugins:maven-release-plugin:update-versions -DgenerateBackupPoms=false
+        git add **pom.xml
```
Comment (maxdanilov) [inline .github/workflows/deploy.yml:104]: We've agreed to rewrite it in a way it works in `sh`

Comment (maxdanilov) [reply .github/workflows/deploy.yml:104]: nvm, this should work already

**Comment by lwille** [comment]
@ThorbenLindhauer (tagging you as the main contributor): as you can see, we're moving from building the community projects on GitHub actions instead of "our" Jenkins.
We made this move to improve transparency as well as provide better feedback for contributors.

I'm going to leave this note in all affected repositories once we're done making the changes.
If you experience issues related to our changes to GitHub Actions, please let us know!

**Pull request merged**